src/client/Microsoft.Identity.Client/Utils/ScopeHelper.cs

```diff
@@ -23,23 +23,31 @@ public static bool ScopeContains(ISet<string> outerSet, IEnumerable<string> poss
             return true;
         }
 
-        internal static SortedSet<string> ConvertStringToLowercaseSortedSet(string singleString)
+        public static HashSet<string> GetMsalScopes(HashSet<string> userScopes)
+        {
+            return new HashSet<string>(userScopes.Concat(OAuth2Value.ReservedScopes));
+        }
+
+        public static HashSet<string> ConvertStringToScopeSet(string singleString)
         {
             if (string.IsNullOrEmpty(singleString))
             {
-                return new SortedSet<string>();
+                return new HashSet<string>();
             }
 
-            return new SortedSet<string>(singleString.ToLowerInvariant().Split(new[] { " " }, StringSplitOptions.None));
+            return new HashSet<string>(
+                singleString.Split(' '), 
+                StringComparer.OrdinalIgnoreCase);
         }
 
-        internal static SortedSet<string> CreateSortedSetFromEnumerable(IEnumerable<string> input)
+        public static HashSet<string> CreateScopeSet(IEnumerable<string> input)
```

Comment:
Can the input be a Readonly collection or equivalent than IEnumerable?

Reply:
I think so, would that help ?

Reply:
yes, as I see you are passing an array as input which is getting converted to IEnumerable and back to Array

Reply:
Unfortunately we have a few public code paths where the input is an IEnumerable.